realm/realm-library/src/main/java/io/realm/OrderedCollectionChangeSet.java

```diff
@@ -34,7 +34,7 @@
     /**
      * State describing the nature of the changeset.
      */
-    public enum State {
+    enum State {
```

Comment:
Hmm, I'm a bit uneasy about this one. These states were used at some point, but maybe that was by QueryBasedSync, which is why Checkstyle says we can remove the public modifier. 

I guess if it isn't being referenced anymore, it is fine.

Reply:
It is just referenced from other classes within the same package, it is safe to change it to package protected.